src/jdns/jdns.c

```diff
@@ -2067,6 +2067,10 @@ int _unicast_do_writes(jdns_session_t *s, int now)
 		}
 
 		// in theory, it is not possible for 'ns' to be null here
+		if (ns == 0)
+		{
+			continue; // but lets make sure it will not stab us in the back
```

Comment:
It's possible only if name_servers->count was update to 0 from another thread. Is it really possible ?
If yes please add a comment how and where.